test-suite/TestSpec.hs

```diff
@@ -283,12 +283,12 @@ spec = do
             it "drops 0" $ buildCell (dropLeft 0 exampleF) `shouldBe` "XXXHelloZthereWYYY"
             it "drops 3" $ buildCell (dropLeft 3 exampleF) `shouldBe` "XXXloZthereWYYY"
             it "drops 8" $ buildCell (dropLeft 8 exampleF) `shouldBe` "XXXZreWYYY"
-            it "drops 20" $ buildCell (dropLeft 20 exampleF) `shouldBe` "XXXZWYYY"
+            it "drops 20" $ buildCell (dropLeft 20 exampleF) `shouldBe` ""
         describe "dropRight" $ do
             it "drops 0" $ buildCell (dropRight 0 exampleF) `shouldBe` "XXXHelloZthereWYYY"
             it "drops 3" $ buildCell (dropRight 3 exampleF) `shouldBe` "XXXHelloZthWYYY"
-            it "drops 8" $ buildCell (dropRight 8 exampleF) `shouldBe` "XXXHeZWYYY"
-            it "drops 20" $ buildCell (dropRight 20 exampleF) `shouldBe` "XXXZWYYY"
+            it "drops 8" $ buildCell (dropRight 8 exampleF) `shouldBe` "XXXHeYYY"
```

Comment:
Is this really desired behavior?  We expect formatting instructions to not be visible from a length perspective.  Could there be a situation where it is desired to have them?

Reply:
It's a good question, and one I've gone back and forth on. I'm happy for you to make the call that the formatting should remain even with empty content.

Reply:
I've changed it so it leaves formatting around empty elements alone. That's certainly the ‘least surprise’ behaviour.